### PR TITLE
frontend/feat: HyperVerge Sdk Integration for Live selfie, Pan and Aa…

### PR DESCRIPTION
**Pull request by ratnadeep-99**
…dhaar flow.

## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [ ] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-common/src/Engineering/Helpers/JBridge.purs [hunk 1]
```diff
@@ -876,4 +881,7 @@ data RouteKeysType  = DEFAULT | RENTAL | ADVANCED
 
 derive instance genericRouteKeysType :: Generic RouteKeysType _
 instance showRouteKeysType :: Show RouteKeysType where show = genericShow
-instance eqRouteKeysType :: Eq RouteKeysType where eq = genericEq
\ No newline at end of file
+instance eqRouteKeysType :: Eq RouteKeysType where eq = genericEq
+
+encodeToBase64Type :: String -> Aff String
+encodeToBase64Type url = makeAff \cb -> encodeToBase64 url (cb <<< Right) $> nonCanceler
```
Comment (Vignesh-772) [inline Frontend/ui-common/src/Engineering/Helpers/JBridge.purs:887]: Add a timeout to this function as it is main flow. Show a toast `Timed Out`

File: Frontend/ui-driver/src/Helpers/Utils.js [hunk 2]
```diff
@@ -555,3 +555,17 @@ export const renderSlider = function (cb) {
     }
   }
 }
+
+export const setActivityResultListener = function (requestCode, cb) {
```
Comment (Vignesh-772) [inline Frontend/ui-driver/src/Helpers/Utils.js:559]: Why requestCode is required?

File: Frontend/ui-driver/src/Screens/RegistrationScreen/Controller.purs [hunk 3]
```diff
@@ -149,15 +172,40 @@ eval (RegistrationAction step ) state = do
           VEHICLE_DETAILS_OPTION -> exit $ GoToUploadVehicleRegistration state step.rcNumberPrefixList
           GRANT_PERMISSION -> exit $ GoToPermissionScreen state
           SUBSCRIPTION_PLAN -> exit GoToOnboardSubscription
-          PROFILE_PHOTO -> exit $ DocCapture state item -- Launch hyperverge
-          AADHAAR_CARD -> exit $ DocCapture state item -- Launch hyperverge
-          PAN_CARD  -> exit $ DocCapture state item -- Launch hyperverge
+          PROFILE_PHOTO -> continueWithCmd state [ pure $ CallHV "ny-selfie-flow" ""]
```
Comment (Vignesh-772) [inline Frontend/ui-driver/src/Screens/RegistrationScreen/Controller.purs:175]: Make it configurable. As All the merchants wont use this.

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

File: Frontend/ui-driver/src/Flow.purs [hunk 4]
```diff
@@ -625,9 +626,38 @@ stopGrpcService =
...
+          let isExpired = runFn1 JB.isSdkTokenExpired expiry
+          if isExpired
+            then cacheNewToken
+          else pure token
+        _ -> cacheNewToken
+  else cacheNewToken
+  where
+    cacheNewToken = do
+      (API.GetSdkTokenResp tokenResp) <- Remote.getSdkTokenBT "86400" API.HyperVerge 
+      let expiry = runFn1 JB.makeSdkTokenExpiry 86400
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Flow.purs:654]: `86400` Add it in constants

File: Frontend/ui-driver/src/Screens/AppUpdatePopUpScreen/Handler.purs [hunk 5]
```diff
@@ -36,7 +40,15 @@ handleAppUpdatePopUp  = do
   act <- lift $ lift $ showScreenWithNameSpace ( AppUpdatePopUpScreen.screen state.appUpdatePopUpScreen)
   _ <- lift $ lift $ doAff $ liftEffect $ terminateUI $ Just "AppUpdatePopUpScreen"
   case act of
-    CD.Accept -> App.BackT $ App.NoBack <$> pure UpdateNow
+    CD.Accept -> do
+      void $ liftFlowBT $ case getMerchant FunctionCall of
+        NAMMAYATRI -> JB.openUrlInApp $ "https://play.google.com/store/apps/details?id=in.juspay.nammayatripartner&pcampaignid=web_share"
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/AppUpdatePopUpScreen/Handler.purs:45]: Add config here

**Review by Rohit4997 — CHANGES_REQUESTED** [review]

File: Frontend/ui-driver/src/Screens/RegistrationScreen/View.purs [hunk 6]
```diff
@@ -694,7 +700,15 @@ getStatus step state =
   where filterCondition item = (state.data.vehicleCategory == item.verifiedVehicleCategory) ||  (isNothing item.verifiedVehicleCategory && item.vehicleType == state.data.vehicleCategory)
 
 dependentDocAvailable :: ST.StepProgress -> ST.RegistrationScreenState -> Boolean
-dependentDocAvailable item state = all (\docType -> (getStatus docType state) == ST.COMPLETED) item.dependencyDocumentType
+dependentDocAvailable item state = 
+  let commonStatus = all (\docType -> statusCompOrManual (getStatus docType state)) item.dependencyDocumentType
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/RegistrationScreen/View.purs:704]: Move this line to last case

File: Frontend/ui-driver/src/Flow.purs [hunk 7]
```diff
@@ -4101,4 +4204,27 @@ getSrcDestConfig state =
       destLon : state.data.activeRide.dest_lon,
       source : state.data.activeRide.source,
       destination : fromMaybe "" state.data.activeRide.destination
-  }
\ No newline at end of file
+  }
+
+
+getHvErrorMsg :: Maybe String -> String
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Flow.purs:4210]: Move this function

**Review by Rohit4997 — CHANGES_REQUESTED** [review]

**Review by Rohit4997 — APPROVED** [review]

**Review by ashishsingh101 — APPROVED** [review]

**Pull request merged**